### PR TITLE
Change Scala Style errors.  Create checkStyleStrict.

### PR DESCRIPTION
**Pull request by schmmd**

### CODE REVIEW

**Comment by schmmd** [comment]
mind doing a release after merging?

File: src/main/resources/allenai-style-config.xml [hunk 1]
```diff
@@ -32,12 +32,14 @@
   <!-- Enabled checks. -->
 
   <!-- Error level checks (strictly not allowed). -->
+  <check level="error" class="org.scalastyle.scalariform.IllegalImportsChecker" enabled="true">
+    <parameters>
+      <parameter name="illegalImports">scala.collection.JavaConversions._</parameter>
```
Comment (jkinkead) [inline src/main/resources/allenai-style-config.xml]: Yay!

File: src/main/resources/allenai-style-config.xml [hunk 2]
```diff
@@ -88,27 +87,37 @@
...
     </parameters>
   </check>
-  <check level="warning" class="org.scalastyle.scalariform.PublicMethodsHaveTypeChecker" enabled="true">
-    <customMessage>Missing return type for public method</customMessage>
-  </check>
   <check level="warning" class="org.scalastyle.scalariform.UppercaseLChecker" enabled="true">
     <customMessage>Prefer using `L` to postfix long literals instead of `l`</customMessage>
   </check>
 
   <!-- Disabled checks, kept here for documentation. -->
```
Comment (jkinkead) [inline src/main/resources/allenai-style-config.xml]: This old comment should probably go below the retired checks?

File: src/main/resources/allenai-style-config.xml [hunk 3]
```diff
@@ -88,27 +87,37 @@
...
 
+  <!-- Retired Checks -->
+
+  <!-- retired because scalariform doesn't remove whitespace from comments. -->
```
Comment (jkinkead) [inline src/main/resources/allenai-style-config.xml]: Comment or TODO to enable if this is fixed?

File: src/main/scala/org/allenai/plugins/StylePlugin.scala [hunk 4]
```diff
@@ -37,25 +39,31 @@ object StylePlugin extends AutoPlugin {
...
-        sourceDir,
-        outputXml,
-        localStreams,
-        0, // How frequently, in hours, to refresh config from URL. Ignored if URL is None.
-        target.value,
-        "/dev/null" // URL cache file. Ignored if URL is None.
+        args = Seq("q"), // "q" for "quiet".
+        config = configFile(target.value), // XML configuration file
+        configUrl = None, // Config URL; overrides configXml.
+        failOnError = false, // If true, the SBT task will treat style warnings as style errors.
```
Comment (jkinkead) [inline src/main/scala/org/allenai/plugins/StylePlugin.scala]: Fix my bad comment? :)

**Comment by jkinkead** [comment]
A few random comments. I'll build after merge!

**Pull request merged**